[MM-45423]: re-added save option to post-dotmenu

michelengelen opened the pull request:
#### Summary
re-added the save/unsave option to the post dotmenu

#### Ticket Link
[MM-45423](https://mattermost.atlassian.net/browse/MM-45423)

#### Related Pull Requests
n/a

#### Screenshots
n/a

#### Release Note
```release-note
NONE
```

matthewbirtch reviewed (approved):
LGTM

jgilliam17 commented:
Apologies, requesting changes after posting approval 🤦🏽‍♀️ - We have a test for Copy menu option that you can use as a guideline `e2e/cypress/tests/integration/messaging/copy_post_text_spec.js`, can you please add something similar for Save? Thanks 🙂

jfrerich reviewed (approved):
LGTM!

michelengelen commented:
@jgilliam17 I have added tests for saving and unsaving a post.

I also did change add a new verification function, moved it to the post.js file and used it in saving spec and the post-pre-header spec (where i got most of the code from.

cc @saturninoabril

jgilliam17 commented:
Thanks @michelengelen for the test. 
Can you please fix failed checks, so we can trigger e2es

jgilliam17 commented on e2e/cypress/tests/integration/messaging/save_post_spec.js:
```diff
@@ -0,0 +1,70 @@
+        });
+    });
+
+    it('MM-Txxxx Save menu item should save post', () => {
```
```suggestion
    it('MM-T4948_1 Save menu item should save post', () => {
```

jgilliam17 commented on e2e/cypress/tests/integration/messaging/save_post_spec.js:
```diff
@@ -0,0 +1,70 @@
+        });
+    });
+
+    it('MM-Txxxx Save hotkey should save post', () => {
```
```suggestion
    it('MM-T4948_2 Save hotkey should save post', () => {
```

jgilliam17 commented on e2e/cypress/tests/integration/messaging/save_post_spec.js:
```diff
@@ -0,0 +1,70 @@
+            verifySavedPost(postId, message);
+
+            // # Remove post from saved by clicking the menu item in the dotmenu
+            cy.uiClickPostDropdownMenu(postId, 'Remove from Saved');
```
Can we add a line here to verify post was unsaved?

jgilliam17 commented on e2e/cypress/tests/integration/messaging/save_post_spec.js:
```diff
@@ -0,0 +1,70 @@
+            verifySavedPost(postId, message);
+
+            // # Press s again to unsave
+            cy.get('body').type('s');
```
same here, can we check if it was unsaved successfully

jgilliam17 reviewed (changes requested):
Adding few test id suggestions

michelengelen commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/223711

jgilliam17 commented on e2e/cypress/tests/integration/messaging/save_post_spec.js:
```diff
@@ -0,0 +1,79 @@
+// [*] indicates an assertion (e.g. * Check the title)
+// Use element ID when selecting an element. Create one if none.
+// ***************************************************************
+
```
```suggestion

// Stage: @prod
```

jgilliam17 reviewed (commented):
Can we please add this tag too? 
Sorry I missed this, I just realized while looking at the report and could't find the new tests there.

michelengelen commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/223871

jgilliam17 commented:
@michelengelen apologies for delay, I'm catching up after time off 
[E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/223871-3147883-onprem-ent-webapp-pr-10701-mattermost/mm-ee-test3147883/mochawesome.html)shows new tests are failing (MM-T4948)

michelengelen commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/229646

jgilliam17 commented:
Checking why cypress run [failed](https://automation-dashboard.vercel.app/cycles/3c1242be-ca8a-4ae3-9029-ca3bdcf58ebf) to run

jgilliam17 commented:
/e2e-test

mattermod commented:
Successfully triggered e2e testing!
https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/229714

jgilliam17 commented:
Thanks @michelengelen - [E2E report ](https://mattermost-cypress-report.s3.amazonaws.com/229714-84be26d-onprem-ent-webapp-pr-10701-mm-ee-test:84be26d/mochawesome.html) looks good.
<img width="1175" alt="Screen Shot 2022-08-15 at 3 45 09 PM" src="https://user-images.githubusercontent.com/52937121/184706322-6db8eacc-4d6d-4b14-aa40-b8343e1ea8f7.png">
